src/tools/debugger/nodes/BreakpointNode.java

```diff
@@ -0,0 +1,57 @@
+package tools.debugger.nodes;
+
+import com.oracle.truffle.api.Assumption;
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.source.SourceSection;
+
+import som.VM;
+import tools.SourceCoordinate;
+import tools.debugger.session.BreakpointEnabling;
+import tools.debugger.session.Breakpoints.MessageBreakpointType;
+
+/**
+ * Node to represent a breakpoint at the AST level.
+ * It has two possible states, enable or disable.
+ */
+public abstract class BreakpointNode extends AbstractBreakpointNode {
+    protected final BreakpointEnabling<?> breakpoint;
+
+    protected BreakpointNode(final SourceSection sourceSection, final MessageBreakpointType type) {
+      switch (type) {
```

Comment:
Please move this out of the constructor to the corresponding code and please don't use a `switch`.